### ISSUE TITLE
verbose logging for getmodule()

### ISSUE THREAD

**Issue by Gaiiden** [KSP-KOS/KOS#797]
if I'm using the part::getmodule() function the KSP log is getting filled with the following lines continuously as I am checking on things every tick:

(Filename: C:/BuildAgent/work/d63dfc6385190b60/artifacts/StandalonePlayerGenerated/UnityEngineDebug.cpp Line: 49)

kOS: Does "MODULERCS" == "MODULETWEAKABLERCS"?
 
(Filename: C:/BuildAgent/work/d63dfc6385190b60/artifacts/StandalonePlayerGenerated/UnityEngineDebug.cpp Line: 49)

kOS: Does "COLLISIONFX" == "MODULETWEAKABLERCS"?
 
(Filename: C:/BuildAgent/work/d63dfc6385190b60/artifacts/StandalonePlayerGenerated/UnityEngineDebug.cpp Line: 49)

kOS: Does "MODULEAGX" == "MODULETWEAKABLERCS"?
 
(Filename: C:/BuildAgent/work/d63dfc6385190b60/artifacts/StandalonePlayerGenerated/UnityEngineDebug.cpp Line: 49)

kOS: Does "FMRS_PM" == "MODULETWEAKABLERCS"?
 
(Filename: C:/BuildAgent/work/d63dfc6385190b60/artifacts/StandalonePlayerGenerated/UnityEngineDebug.cpp Line: 49)

kOS: Does "KASMODULEGRAB" == "MODULETWEAKABLERCS"?
 
(Filename: C:/BuildAgent/work/d63dfc6385190b60/artifacts/StandalonePlayerGenerated/UnityEngineDebug.cpp Line: 49)

kOS: Does "KOSNAMETAG" == "MODULETWEAKABLERCS"?
 
(Filename: C:/BuildAgent/work/d63dfc6385190b60/artifacts/StandalonePlayerGenerated/UnityEngineDebug.cpp Line: 49)

kOS: Does "MODULETWEAKABLERCS" == "MODULETWEAKABLERCS"?
 
(Filename: C:/BuildAgent/work/d63dfc6385190b60/artifacts/StandalonePlayerGenerated/UnityEngineDebug.cpp Line: 49)

kOS: yes it does

**Comment by hvacengi** [comment]
I have another couple of spots where I've noticed some log spam when doing things every tick (orbitable:position namely).  I'll add this to the pull request I'm composing.

**Comment by Gaiiden** [comment]
This is still present and actually has caused noticeable performance impacts, especially if I do something like this:

`until avionics:getmodule("modulescienceexperiment"):hasevent("review telemetry") and science:getmodule("modulescienceexperiment"):hasevent("reset meteorological data") {}.`

Which is the program entering a wait state until it sees the user has saved data from two experiments the script triggered. FPS dropped to a very noticeable stutter

**Comment by hvacengi** [comment]
For various perfomrnace reasons (including potential log spamming) you should avoid calling `getmodule` over and over again, since the code has to walk the tree to find the module every single time.  The logging still needs to get changed, but I'd recommend using something like
```
set avionicsSciExp to avionics:getmodule("modulescienceexperiment").
wait until avionicsSciExp:hasevent("review telemetry").
```

I never did make a PR on the logging, it's still on my todo list to audit all of those calls.

**Comment by Gaiiden** [comment]
good point on not using `getmodule` constantly. That may in fact be causing more of a slowdown than the logging, I didn't think of that. Although at the same time I've seen similar performance hits just from logging as well which is why I thought of that first
